### DISCUSSION TITLE
Sign transaction (On near blockchain) using Trust Wallet Core . WASM

### DISCUSSION THREAD

**Discussion by vlmoon99** [trustwallet/wallet-core#3001] [Q&A] [answered]
This is my function ->

// fromAddress (String of Account id)
// toAddress (String of Account id)
// transferAmount (String with numbers of amount (in yoctoNear) )
// privateKey (String of base 64 encoded private key)
// nonce (String of with number of nonce taking from the account)
```
function signTransactionNear(
  fromAddress,
  toAddress,
  transferAmount,
  privateKey,
  nonce
) {
  try {
    const { TW, WalletCore } = window;
    const { AnySigner, CoinType } = WalletCore;
    const amount = new BN(transferAmount);
    const amountBytes = amount.toArrayLike(Uint8Array, "be", 16).reverse();
    const input = TW.NEAR.Proto.SigningInput.create({
      signerId: fromAddress,
      nonce: Long.fromString(nonce),
      receiverId: toAddress,
      actions: [
        {
          transfer: TW.NEAR.Proto.Transfer.create({
            deposit: amountBytes,
          }),
        },
      ],
      privateKey: Uint8Array.from(Buffer.from(privateKey, "base64")),
    });
    const encoded = TW.NEAR.Proto.SigningInput.encode(input).finish();

    const outputData = AnySigner.sign(encoded, CoinType.near);

    const output = TW.NEAR.Proto.SigningOutput.decode(outputData);

    return JSON.stringify(output);
  } catch (error) {
    console.error(JSON.stringify({ error: error.message }));
    return JSON.stringify({ error: error.message });
  }
}

```
After I execute this function I got the result - >
`{"signedTransaction":"QAAAAGYxYmFmNDJmMWI4YjIxZWIzYWQ0OTUzMDQyMmMzNmM5MGU4NGFmZGJiOTY2NTYwYzRkN2NlMjI1ZjgzNjM0ZWMA8br0LxuLIes61JUwQiw2yQ6Er9u5ZlYMTXziJfg2NOzBFZjsGW0AABUAAABmbHV0dGVyd2FsbGV0LnRlc3RuZXQBAAAAA/////////9/AAAAAAAAAAAATG0dHH305WBqcHxveY2aa8HHpveFNkTd4LAlcMjXEYpswfUFDKo13HpRAP5C7Y+qjPAwqXtVdgrzpuuZ1qMCCQ=="}}`

Bu when I sended it on the blockchain I have the next result -> 
` ({jsonrpc: 2.0, error: {name: REQUEST_VALIDATION_ERROR, cause: {name: PARSE_ERROR, info: {error_message: Failed to decode transaction: Unexpected variant index: 154}}, code: -32700, message: Parse error, data: Failed to decode transaction: Unexpected variant index: 154}, id: dontcare})`

Whats wrong with my function. What I need to do to resolve my issue ?

**Comment by frol** [comment, marked as answer]
One obvious thing that is missing is a recent `block_hash`, which has to be specified in the transaction.
